Sync to CICE-Consortium/main (2025-01-23)

NickSzapiro-NOAA opened the pull request:
For detailed information about submitting Pull Requests (PRs) to the CICE-Consortium,
please refer to: <https://github.com/CICE-Consortium/About-Us/wiki/Resource-Index#information-for-developers>


## PR checklist
- [x] Short (1 sentence) summary of your PR: 
    Merge authoritative CICE-Consortium:main into EMC fork, including baseline changes in outgoing lw for history and coupling and Icepack related to ponds
- [x] Developer(s): 
    Tony Craig, Dave Bailey, David Clemens-Sewall, Nick Szapiro
- [ ] Suggest PR reviewers from list in the column to the right.
- [x] Please copy the PR test results link or provide a summary of testing completed below.
    UFS regression testing, see https://github.com/ufs-community/ufs-weather-model/pull/2569
- How much do the PR code changes differ from the unmodified code? 
    - [ ] bit for bit
    - [ ] different at roundoff level
    - [x] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [x] No
- Does this PR update the Icepack submodule?  If so, the Icepack submodule must point to a hash on Icepack's main branch.
    - [x] Yes
    - [ ] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [x] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [x] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [x] No 
- [x] Please document the changes in detail, including _why_ the changes are made.  This will become part of the PR commit log.

EMC/CICE sync to authoritative upstream with baseline changes. Closes https://github.com/NOAA-EMC/CICE/issues/98

apcraig committed:
Update Perlmutter Port, switch to cce/17.0.0, gnu/12.3, and intel/2023.2.0 (#1000)

Update Icepack to #fa56e7e387b1, needed for cce/17.0.0

Cleaned up some trailing blanks.

dabail10 committed:
Fix outgoing lw for history and coupling. (#999)

This is a two-line change in ice_flux.F90 (scale_fluxes) to check for the condition where aice > 0, but flwout > -puny (effectively zero). This means the cell was ice free at the beginning of the step, but non-zero at the end. It sets the outgoing longwave to the open ocean value based on the freezing temperature. This is bfb, but will change the history output of flwup and flwup_ai. Test results are coming.

apcraig committed:
Update Icepack to #43ead56380b, hocn and flushing velocity issue, not bfb (#1002)

Update Icepack to #43ead56380b, hocn and flushing velocity issue, not bfb.

See CICE-Consortium/Icepack#504

NickSzapiro-NOAA committed:
Enable triggering restart writes at any forecast time ("restart_fh") via UFS configuration as for other components (#1001)

* No ncat,ntrcr in call icepack_aggregate in ice_prescribed_mod.F90

* Enable writing restarts that can be triggered at any forecast time ("restart_fh") via UFS configuration as for other components

Part of https://github.com/ufs-community/ufs-weather-model/issues/2348

* Add 1 more #ifndef CESMCOUPLED to avoid unused variable warning for cesm

* #ifndef for dtime here too

apcraig committed:
Update derecho gnu to add coverage capability with lcov (#1003)

Update derecho gnu to support code coverage testing

    Update to ncarenv/24.12 and gcc/12.4.0
    Add coverage compiler flags
    Update lcov scripting as needed

As part of the CICE port to the new coverage tools, a bug was found in the tools that requires debug flags to be disabled when compiling with coverage flags. This was not the case when coverage was running on cheyenne. See linux-test-project/lcov#385

NickSzapiro-NOAA committed:
Merge 'CICE-Consortium/main' into EMC/develop

DeniseWorthen commented:
@NickSzapiro-NOAA I see you mention that the Icepack update is not B4B. But the lwout change---wouldn't that also change answers?

NickSzapiro-NOAA commented:
@DeniseWorthen Yes, thank you. I've updated summary. I removed my baselines for disk space but will take another look at cpld_control_gfsv17 RT for sanity check

NickSzapiro-NOAA commented:
The results are somewhat weird/confusing and get into if cice gridcell has ice and if fluxes are per unit ice area

Copying https://github.com/CICE-Consortium/CICE/issues/67#issue-286488502 as helpful: 
"flwup in history is flwout elsewhere in the code, and in the code calculations this is the value only over sea ice. However it’s later divided by aice for the cesm coupler, and that happens before it’s sent to history. flwup_ai in history is flwout*aice, so it’s back to the only-over-ice value."

We don't currently output flwup_ai. Here is flwup_1 from the daily-averaged timestep output for current baseline (left) and PR update (right), with a colorbar that goes [-1000, -100] W/m2 on purpose:
![flwup_baseline_ciceSync_1day](https://github.com/user-attachments/assets/a3978fa8-7728-4f8d-ba58-ff63eb8153a2)
to highlight that for this PR (right) there are "blue outlines" where flwup<-1000 W/m2. These are associated with low aice (like 1E-6). 

In FV3, ulwrf doesn't show these and diffs are ~O(10 W/m2) after 24 h. 
I've poked around and it looks a bit weird but seems ok

Note that there is still open issue on what do in CICE for coupling when aice comes from or goes to zero

DeniseWorthen commented:
I had a hard time following the exact back/forth on Consortium, but of course since we're using the same cap, we also divide by aice before exporting (scale_fluxes). 

I've known about the difference between f()_ai and just f(). It's my understanding that f()_ai is the time averaged value of the ai-weighted field; that won't be the same as mean(f)*mean(ai). I think this is an issue when they decided to save only daily output f() and ai() for CICE but I don't think I was able to convince anyone.

FernandoAndrade-NOAA commented:
Testing on [2569](https://github.com/ufs-community/ufs-weather-model/pull/2569) is complete, please continue with merging, thank you.
